### PR TITLE
✨  Mt 1215 dialogbox

### PR DESCRIPTION
**Pull request by miloszhygen**

### CODE REVIEW

File: src/svelte/DialogBox.svelte [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<script lang="ts">
+  export let dialogBoxClosed: boolean = false
+  export let closeBtnLabel: string = ''
+
+  function handleClose() {
+    dialogBoxClosed = true
+  }
+</script>
+
+<div class={dialogBoxClosed ? 'dialog-box--closed' : 'dialog-box'} data-testid="dialog-box">
```
Comment (lha047) [inline src/svelte/DialogBox.svelte:10]: Challenge :P Try to limit the use of divs

File: src/svelte/DialogBox.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { render } from '@testing-library/svelte'
+import DialogBox from './DialogBox.svelte'
+
+describe('DialogBox', () => {
+
+  test('Renders when dialogBoxClosed is false', () => {
+    const componentOptions = {
+      dialogBoxClosed: false,
+    }
+    const { getByTestId } = render(DialogBox, componentOptions)
```
[the remark this answers is not in the document]
Comment (miloszhygen) [reply src/svelte/DialogBox.test.ts:13]: The component does not contain any text to query :)

File: src/svelte/DialogBox.test.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const dialogBox = getByTestId('dialog-box')
+    expect(dialogBox.classList[0]).toEqual('dialog-box')
+  })
+  test('Is hidden when dialogBoxClosed is true', () => {
+    const componentOptions = {
```
[the remark this answers is not in the document]
Comment (miloszhygen) [reply src/svelte/DialogBox.test.ts:18]: Same as above ☝️

File: src/scss/components/_dialog-box-closable.scss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  height: auto;
+  border: $border-width-regular solid $color-primary;
+  &--closed{
+    display: none;
+    visibility: hidden;
+  }
+
+  &--actions{
+    position: relative;
+    background-color: green;
```
[the remark this answers is not in the document]
Comment (miloszhygen) [reply src/scss/components/_dialog-box-closable.scss]: removed :)

File: src/svelte/DialogBox.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lha047) [reply src/svelte/DialogBox.test.ts:13]: If you put the aria-label which looks to be required. There should be a label to query ;)

File: src/scss/components/_dialog-box-closable.scss [hunk 5]
```diff
@@ -0,0 +1,43 @@
+@use '../theme' as *;
+@use '../utilities/mixins' as *;
+@use '../color-tokens' as *;
+
+.dialog-box {
+  background-color: $color-mt-lighter-light-blue;
```
Comment (lha047) [inline src/scss/components/_dialog-box-closable.scss:6]: Its possible to use css variables on colors as well. See other places too

File: src/scss/components/_dialog-box-closable.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+.dialog-box {
+  --min-clickable-width-height: 2.75rem; // 44px
+  display: block;
+  background-color: var(--color-mt-lighter-light-blue);
+  width: 100%;
+  height: auto;
+  border: var(--border-width-regular) solid var(--color-primary);
+  position: relative;
+
```
Comment (lha047) [inline src/scss/components/_dialog-box-closable.scss:13]: Need focus styling

File: src/svelte/DialogBox.svelte [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  function handleClose() {
+    dialogBoxClosed = true
+  }
+</script>
+
+<div
+  class={dialogBoxClosed ? 'dialog-box--closed' : 'dialog-box'}
+  data-testid="dialog-box"
+  role="dialog"
```
Comment (lha047) [inline src/svelte/DialogBox.svelte:13]: Missing `aria-label` or `aria-labelledby`. `role=dialog` should have a label. If you look in the storybook addon for a11y you see the a11y error. 

Maybe heading should be passed to the component? And then use `aria-labelledby`

**Review by lha047 — CHANGES_REQUESTED** [review]
Found some more a11y things :)

File: src/svelte/DialogBox.svelte [hunk 8]
```diff
@@ -11,13 +12,15 @@
...
-  tabindex="0">
+  tabindex="0"
+  aria-labelledby="title">
   <button
     data-testid="dialog-box-close"
     class="button button--link dialog-box--button"
     on:click={handleClose}
     aria-label={closeBtnLabel} />
   <div class="dialog-box--content">
+    <p id="title">{dialogBoxTitle}</p>
```
Comment (lha047) [inline src/svelte/DialogBox.svelte]: This should be a heading. I guess h2. But ask Malin.

Comment (lha047) [inline src/svelte/DialogBox.svelte]: The id should be something unique. There is a chance `id="title"` occurs on a page. I suggest using something like:
```
<script context="module" lang="ts">
  let instanceCounter = 0
</script>
<script lang="ts">
  const dialogBoxHeadingId = `ui-dialog-box-${instanceCounter++}`
</script>
```
There is an example in `Select.svelte`

File: src/scss/components/_dialog-box-closable.scss (hunk 6 above)
Comment (lha047) [reply src/scss/components/_dialog-box-closable.scss:13]: Still need focus styling

**Review by lha047 — CHANGES_REQUESTED** [review]
Left some comments about a11y and focus style

File: src/scss/components/_dialog-box-closable.scss (hunk 6 above)
Comment (miloszhygen) [reply src/scss/components/_dialog-box-closable.scss:13]: could I use `@include focus-outline();` ?

File: src/svelte/DialogBox.svelte [hunk 9]
```diff
@@ -0,0 +1,37 @@
+<script context="module" lang="ts">
+  let instanceCounter = 0
+</script>
+
+<script lang="ts">
+  export let dialogBoxClosed: boolean = false
```
Comment (lha047) [inline src/svelte/DialogBox.svelte:6]:
```suggestion
  export let isOpen: boolean = true
```
The closed = false got me abit confused. Felt its easier to read using the true value.
I prefer to prefix booleans with `is`. Dont think you need `dialogBox` its already in the component name.

File: src/svelte/DialogBox.svelte [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  export let dialogBoxTitle: string = ''
```
Comment (lha047) [inline src/svelte/DialogBox.svelte:7]:
```suggestion
  export let title: string = ''
```

File: src/svelte/DialogBox.svelte [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  export let closeBtnLabel: string = 'Lukk'
```
Comment (lha047) [inline src/svelte/DialogBox.svelte:8]:
```suggestion
  export let closeBtnAriaLabel: string = 'Lukk'
```
To make it explisit

File: stories/components/DialogBox.stories.svelte [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+  <div class="tags-wrapper">
+    <DialogBox {dialogBoxTitle} {dialogBoxClosed} {closeBtnLabel}>
+      <p>Dialogboks innhold</p>
+    </DialogBox>
+  </div>
+
+  <hr />
+  <h2>Enkel</h2>
+  <div class="tags-wrapper">
+    <DialogBox {dialogBoxClosed} {closeBtnLabel}>
```
Comment (lha047) [inline stories/components/DialogBox.stories.svelte:31]: It should have been a way to enforce title of the box. Voice over reads "Lukk dialogboks" as the title of the dialog.

File: src/svelte/DialogBox.svelte [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+</script>
+
+<div
+  class={dialogBoxClosed ? 'dialog-box--closed' : 'dialog-box'}
+  data-testid="dialog-box"
+  role="dialog"
+  tabindex="0"
+  aria-labelledby={dialogBoxHeadingId}>
+  <h2 id={dialogBoxHeadingId} class="h4">
+    <button
```
Comment (lha047) [inline src/svelte/DialogBox.svelte:24]: Could the button be after the title in the DOM order? It feels weird when Voice Over reads "Look dialogboks" before the title.

**Review by lha047 — CHANGES_REQUESTED** [review]
Nice job on the focus and styling

There is still some issues with screen reader I think. Also have some suggestions on readability

File: src/svelte/DialogBox.svelte (hunk 9 above)
Comment (miloszhygen) [reply src/svelte/DialogBox.svelte:6]: Good point 👍

File: stories/components/DialogBox.stories.svelte (hunk 12 above)
Comment (miloszhygen) [reply stories/components/DialogBox.stories.svelte:31]: Hmm... yes, this is unfortunate, I'll see if its possible to fix the order of what is read first

Comment (miloszhygen) [reply stories/components/DialogBox.stories.svelte:31]: I have changed the order of the elements :)

File: src/scss/components/_feedback-container.scss [hunk 14]
```diff
@@ -2,17 +2,19 @@
 
 .feedback-container {
   --width-outside-content: 100px;
```
Comment (lha047) [inline src/scss/components/_feedback-container.scss:4]: Can this be defined in rem?

File: src/scss/components/_feedback-container.scss [hunk 15]
```diff
@@ -2,17 +2,19 @@
...
-  --feedback-from-top: 45vh;
-  --feedback-box-width: 256px;
+  --feedback-from-top: 20vh;
+  --feedback-box-width: 260px;
```
Comment (lha047) [inline src/scss/components/_feedback-container.scss:6]: Can this be defined in rem?

File: src/scss/components/_dialog-box-closable.scss [hunk 16]
```diff
@@ -0,0 +1,49 @@
+@use '../theme' as *;
+@use '../utilities/mixins' as *;
+@use '../color-tokens' as *;
+
+.dialog-box {
+  @include focus-outline();
+  --focus-box-shadow: 0 0 0 2px var(--color-mt-background),
```
Comment (lha047) [inline src/scss/components/_dialog-box-closable.scss:7]: Can this be defined in rem?

**Review by lha047 — APPROVED** [review]
Nice work 🎉 

Left some comments about rem vs px.

**Pull request merged**